### ISSUE TITLE
Cancel G Suite automatically when domain is canceled or transferred

### ISSUE THREAD

**Issue by Lochlaer** [Automattic/wp-calypso#26132]
tl;dr Ideally, our systems should automatically cancel (i.e., remove) the G Suite upgrade from the user's Store Admin and account when the domain attached to it is canceled or transferred away. Refunds can be issued if the cancelation is done within 30 days of purchase. 

<h4>DESCRIPTION OF ISSUE</h4>

**CANCELATIONS**

When a user cancels a domain that also has G Suite attached to it, they often neglect to cancel the G Suite upgrade as well. If the domain cancelation goes through, eventually the G Suite account does stop working once all the appropriate DNS records are removed, but in the meantime the active subscription might cause some issues. For example, if it's still on their account at the time of renewal and it's set to auto-renew, they'll automatically get charged for it. 

**TRANSFERS**

When a user transfers their domain away, they will need to move their MX records to their new domain registrar/host in order for their G Suite to continue working. Even if they were to successfully do that, it's still a less-than-ideal situation to keep their G Suite sub with us because any time they want to, say, add a new license to their account, we would need to manually add it via GARC or the GRC since the user is no longer accessing their G Suite via the WPCOM Domains page. 

**Canceling the sub and transferring the account back to Google is the best solution for both scenarios above.** 

<h4>IMPORTANT NOTE</h4>

Canceling a G Suite sub (removing it from the Store Admin) does not delete or otherwise deactivate the G Suite account itself. Instead, it is automatically transferred back to Google for management and billing almost immediately. The user then gets an email message from Google notifying them that the G Suite account for the domain is no longer managed by the reseller (i.e., us) but is instead managed by Google, and to direct any questions about it to Google. The user can ignore that message as the account will eventually be suspended/deactivated by Google once the domain stops working.

**Comment by Lochlaer** [comment]
@matthusby Not sure if this would fit into your work schedule right now, but just in case I though I'd bump it up. :-) No worries if it's not prioritized at this time!

**Comment by Lochlaer** [comment]
@klimeryk I know it's not a priority, but if you have any bandwidth to do this, it would be most appreciated! :-)

**Comment by klimeryk** [comment]
Heh, this is a fun one. Unfortunately, it looks like a complicated one - we'd need to make changes on the frontend to ensure the user is aware of the implications of canceling a domain. Just so that we're on the same page: the email will continue working, right? Until the renewal time of G Suite (the account is moved to Google at cancelation time, but continues working - as long as the domain itself works as well, of course).

That would at least make it a bit simpler to explain in the UI (and make it less alarming).

To help prioritize this better, what would you say is the impact of this in terms of number of affected users, @Lochlaer? Do we get chargebacks for these renewals? (cc @rralian since this seems to more affect the payment side of things than domains)
